File: Additional practice problems/Java/01_Average Permutation.java
import java.util.*;
import java.lang.*;
import java.io.*;

class Codechef
{
	public static void main (String[] args) throws java.lang.Exception
	{
		// your code goes here
		Scanner sc = new Scanner(System.in);
		int t = sc.nextInt();
		
		for(int i =0 ; i< t ; i++)
		{
		    int n = sc.nextInt();
		    
		    permutation(n);
		    
		}

	}
	
	
	public static void  permutation(int n)
	{
	    
	    if(n==3)
	    {
	        System.out.println("1 2 3");
	        return;
	    }
	    
	    System.out.print(n + " " + (n-2) + " ");
	    
	    for(int i =1 ; i<=n-4 ; i++ )
	    {
	        System.out.print(i + " ");
	    }
	    
	    
	    System.out.println( (n-3) + " " + (n-1) );
	    
	    
	    

	}
	
	
}

File: Additional practice problems/01_Average Permutation.py
# cook your dish here


t = int(input())

for _ in range(t):
    
    s = int(input())
    
    if(s==3):
        print("1 2 3")
        continue
    
    a =[]
    
    a= a+ [s , s-2]
    
    for i in range(1 , s-3 ):
        a.append(i)
    
    a = a+ [s-3 , s-1] 
    
    print(*a)
    
    
    
    
    
    
    
    
    